api/v1alpha1/jobset_webhook.go

```diff
@@ -66,7 +70,15 @@ func (r *JobSet) ValidateCreate() error {
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *JobSet) ValidateUpdate(old runtime.Object) error {
-	return nil
+	oldObj := old.(*JobSet)
+	var allErr []error
+	for index := range r.Spec.ReplicatedJobs {
+		oldObj.Spec.ReplicatedJobs[index].Template.Spec.Template.Spec.NodeSelector = r.Spec.ReplicatedJobs[index].Template.Spec.Template.Spec.NodeSelector
+		if !reflect.DeepEqual(oldObj.Spec.ReplicatedJobs, r.Spec.ReplicatedJobs) {
+			allErr = append(allErr, fmt.Errorf("the spec.ReplicatedJobs value is imutable excepting Spec.ReplicatedJobs[index].Template.Spec.Template.Spec.NodeSelector"))
```

Comment:
```suggestion
			allErr = append(allErr, fmt.Errorf(fmt.Sprintf("spec.ReplicatedJobs updates may not change fields other than %s\n%v", strings.Join(updatableFields, ","))
```

and define this above:
```
var updatableFields = []string{
	"`template.spec.template.spec.nodeSelector`",
}
```